ast/src/asm_analysis/display.rs

```diff
@@ -47,13 +47,17 @@ impl<T: Display> Display for AnalysisASMFile<T> {
                 Item::Machine(machine) => {
                     write_indented_by(f, format!("machine {name}{machine}"), current_path.len())?;
                 }
-                Item::Expression(ExpressionWithTypeName { e, type_name }) => write_indented_by(
+                Item::Expression(ExpressionWithTypeScheme { e, type_scheme }) => write_indented_by(
                     f,
                     format!(
-                        "let {name}{} = {e};\n",
-                        type_name
+                        "let{} {name}{} = {e};\n",
```

Comment:
This is done >=3 times

Reply:
Fixed in https://github.com/powdr-labs/powdr/pull/1083